src/shared/ui/Input/ui/index.tsx

```diff
@@ -23,15 +32,39 @@ const Input = (props: IInputCommonProps) => {
     onFormChange(value)
   }
 
+  const onIconClick = () => {
+    if (isPasswordType === "password") {
+      setIsPasswordType("text")
+    } else {
+      setIsPasswordType("password")
+    }
+  }
+
   return (
-    <Form.Control
-      type={type}
-      name={name}
-      placeholder={placeholder}
-      value={value}
-      onChange={onInputChange}
-      className={classNames("mb-3", className)}
-    />
+    <div className={classNames("input-with-icon-wrap")}>
+      <Form.Control
+        type={isPasswordType ? isPasswordType : type}
+        name={name}
+        placeholder={placeholder}
+        value={value}
+        onChange={onInputChange}
+        className={classNames("default-input", className, { "input-error-shadow": error })}
+      />
+
+      {isPasswordInput && isPasswordType === "password" && (
```

Comment:
Please create an "icon" or "Icon" prop that accepts JSX.Element. Then you can pass there whatever icon you want and render it right here.

Reply:
Done